contrib/pkg/sets/sets.go

```diff
@@ -1,36 +1,45 @@
 package sets
 
 import (
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sync"
+
+	"github.com/rotisserie/eris"
+	"github.com/solo-io/skv2/pkg/ezkube"
 	"k8s.io/apimachinery/pkg/util/sets"
 )
 
+var NotFoundErr = func(resourceType ezkube.ResourceId, id ezkube.ResourceId) error {
+	return eris.Errorf("%T with id %v not found", resourceType, Key(id))
+}
+
 // k8s resources are uniquely identified by their name and namespace
-func Key(objectMeta metav1.Object) string {
-	return objectMeta.GetName() + "." + objectMeta.GetNamespace()
+func Key(id ezkube.ResourceId) string {
+	return id.GetName() + "." + id.GetNamespace() + "." + id.GetClusterName()
```

Comment:
This should most likely omit the ClusterName if it is not available

Reply:
i was thinking the same, but then the same could apply to Namespace (since it's empty on cluster scoped objects). my preference is to leave it the way it is, as it's only used as a key in a Set